vm/class.go

```diff
@@ -60,12 +61,16 @@ type BaseClass struct {
 	Methods *environment
 	// ClassMethods contains this class's methods
 	ClassMethods *environment
-	// SuperClass points to the class it inherits
-	SuperClass *RClass
+	// pseudoSuperClass points to the class it inherits
+	pseudoSuperClass *RClass
+	// This is the class where we should looking for a method.
+	// It can be normal class, singleton class or a module.
+	superClass *RClass
```

Comment:
In Ruby, module is almost like class internally and module including represents as inheritance, so we need to know the actual superclass (might be a module), and the superclass for checking class inheritance. For example:
```ruby
module Foo
end

class Baz
end

class Bar < Baz
  include(Foo)
end
```

In the internal, the inheritance looks like `Bar < Foo < Baz` (record is `superClass` attribute)
But we need to know the class inheritance `Bar < Baz` so we also need `pseudoSuperClass`